metadata/ota_metadata/metadata_gen.py

```diff
@@ -147,12 +193,27 @@ def gen_metadata(
             stat = os.stat(os.path.join(target_dir, d))
             nlink = stat.st_nlink
             inode = stat.st_ino if nlink > 1 else ""
+            sha256hash = _file_sha256(os.path.join(target_dir, d))
+
+            # if compression is enabled, try to compress the file here
+            compress_alg = ""
+            if compression_enabled:
+                if zstd_compress_file(
+                    cctx,
+                    os.path.join(target_dir, d),
+                    os.path.join(compressed_dir, sha256hash),
+                    cmpr_ratio=cmpr_ratio,
+                    filesize_threshold=filesize_threshold,
+                ):
+                    compress_alg = f",{ZSTD_COMPRESSION_ALG}"
+
             regular_list.append(
                 f"{_join_mode_uid_gid(target_dir, d, nlink=True)},"
-                f"{_file_sha256(os.path.join(target_dir, d))},"
+                f"{sha256hash},"
                 f"{_encapsulate(d, prefix=prefix)},"
                 f"{size},"
                 f"{inode}"
+                f"{compress_alg}"  # ensure the compress_alg is at the end
```

Comment:
`,` should be required.
```suggestion
                f"{inode},"
                f"{compress_alg}"  # ensure the compress_alg is at the end
```

Reply:
I pre-add the `,` in the <compress_alg>(like this `,zstd`) as the <compress_alg> is optional.
It is OK to have a `,` at the end of the regular meta line without <compress_alg> field set like following case?
```
# bare comma at the end of line
0644,1000,1000,1,0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef,'path/to/file',1234,12345678,
```

Reply:
need `,` if compress alg field is empty.
If we see `a,b,c,`(end of comma) , we can find that `d` exists as empty, but if d is empty and the field is `a,b,c`, we can't determine if `d`(=4th) filed exits or not.

Reply:
I'd like to introduce this manner.